colcon_core/package_identification/__init__.py

```diff
@@ -164,3 +180,10 @@ def _identify(extensions_same_prio, desc):
         "Package '{desc.path}' with type '{desc.type}' and name '{desc.name}'"
         .format_map(locals()))
     return desc
+
+
+def _are_descriptors_equal(desc1, desc2):
```

Comment:
Why not use the existing `__eq__` definition? I notice it only uses `path`, `type`, and `name` slots, maybe it should be updated to include the remaining slots?

Reply:
The `__eq__` only checks if the object represents the same package (which is uniquely identifies by these three attributes). If additional dependencies and metadata are added they are not used for that comparison. Therefore I needed a separate comparison covering all slots.